### PR TITLE
Add an API to calculate payouts when the taker goes short

### LINKED ISSUE #1432: Add API to generate a payout curve to allow the taker to go short

### PR DESCRIPTION
**Pull request by luckysori**
Fixes #1432.

### CODE REVIEW

**Review by rishflab — APPROVED** [review]

File: model/src/payout_curve.rs [hunk 1]
```diff
@@ -158,13 +182,21 @@ struct PayoutParameter {
...
-    fn into_payouts(self) -> Result<Vec<Payout>> {
+    fn into_payouts_long_taker(self) -> Result<Vec<Payout>> {
         generate_payouts(
             self.left_bound..=self.right_bound,
             bitcoin::Amount::from_sat(self.short_amount),
             bitcoin::Amount::from_sat(self.long_amount),
         )
     }
+
+    fn into_payouts_short_taker(self) -> Result<Vec<Payout>> {
```
Comment (thomaseizinger) [inline model/src/payout_curve.rs]: Why is this necessary? If the caller know the role of the position, wouldn't that be good enough?

Comment (luckysori) [reply model/src/payout_curve.rs]: It is not necessary but convenient. Before we had an `into_payouts` function which knew that the taker was always long. 

I can remove both functions and inline this logic.

Comment (thomaseizinger) [reply model/src/payout_curve.rs]: Shouldn't we change `maia` to care about `long` and `short` instead of `maker` and `taker`?

I am happy to merge this for now but we should fix this in maia I believe.

Comment (thomaseizinger) [reply model/src/payout_curve.rs]: I created a ticket for it: https://github.com/comit-network/maia/issues/16

Comment (luckysori) [reply model/src/payout_curve.rs]: > Shouldn't we change `maia` to care about `long` and `short` instead of `maker` and `taker`?

Ultimately yes.

File: model/src/payout_curve.rs [hunk 2]
```diff
@@ -43,29 +41,63 @@ mod utils;
...
+/// * long_leverage: leverage used by the party with the long position
+/// * short_leverage: leverage used by the party with the short position
+/// * n_payouts: number of segments into which the payout curve is discretised
+/// * fee: offset applied to the curve representing a fee paid between parties
 ///
 /// ### Returns
 ///
 /// The list of [`Payout`]s for the given price, quantity and leverage.
-pub fn calculate(
+pub fn calculate_long_taker(
```
Comment (da-kami) [inline model/src/payout_curve.rs]: Is the expectation that there will be a `calculate_short_taker` as well? 
Will there be separate functions for `calculate_long_maker` and `calculate_short_maker`? 

I would actually find it easier if we would move this logic on the `Cfd`. The cfd knows position and role and can decide how to call the `calculate_payout_parameters` parameters. I find the indirection with `calculate_long_taker` in between confusing.

**Review by klochowicz — APPROVED** [review]
I'm happy for this to go in, as it's an improvement. removing the "maker" and "taker" from this part of the code should probably be tackled along with `maia` refactor.

File: model/src/payout_curve.rs (hunk 2 above)
Comment (luckysori) [reply model/src/payout_curve.rs]: Fair, I can try to move these APIs to the `model` crate.

**Review by thomaseizinger — APPROVED** [review]
